docs/03-Filtering Patches.md

```diff
@@ -1,18 +1,51 @@
 ## Filtering Patches
 
-You can ignore local changes by setting `Palindrom.filterLocalChange` function. This function should accept a JSON Patch operation and either return it or return a falsy value. Obviously, returning a falsy value will result an ignored operation.
+### 1. By defining non-enumerable properties _(recommended)_:
+Assuming your Palindrom instance is instantiated and your `palindrom.obj` is ready, you can create a non-enumerable object child in `palindrom.obj` and use it as a local non-synced store for your needs.
+
+#### Example:
+
+```js
+Object.defineProperty(palindrom.obj, "myLocalNonSyncedObject", {
+    enumerable: false,
+    value: {},
+    writable: true
+});
+```
+Now `palindrom.obj.myLocalNonSyncedObject` is an object this is not synced with the server. This is ideal for keeping client-side related information (eg: div height, or something like `isModalWarningDisplayed = true`).
+
+Example:
+
+```js
+palindrom.obj.myLocalNonSyncedObject.divHeight = 50;
+```
+... is a local change.
+
+**Note:  The added property must be an object.***
+
+_*Adding a non-enumerable property adds a property to the object in a non-synced manner, however, if you modify this property Palindrom will nonetheless try to sync it. To overcome that, the added property must be an object that doesn't change itself, only its children do, and that doesn't trigger Palindrom._
```

Comment:
Doesn't `writeable: false` fix it?

```js
"use strict";
Object.defineProperty(palindrom.obj, "myLocalNonSyncedObject", {
    enumerable: false,
    value: {},
    writable: false
});
```

See "Writable attribute" at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty

Reply:
It fixes it, but what's the point of adding a variable that is constant? I guess in most cases it is an oxymoron. 

Maybe my explanation sucks but I tried my best to word it and it's just hard to explain. At least for me. 

Maybe you can word it? I'll explain in code:

```js
var obj = {};
var proxy = profixy(obj);
proxy.prop = 1;
```

In this case `proxy` is observed. If you say `proxy.prop = x`, you are not modifying `prop`, you're modifying `proxy` right? That's why even if `prop` is non-enumerable, the proxy will be triggered. 

That's why you have to add `prop` that you will not change (an object reference). Like:

```js
var obj = {};
var childObj = {}; // non-enumerable
var proxy = profixy(obj);
proxy.prop = childObj;
```

Now if you set `proxy.prop.a = 1`, you're not modifying a proxy, rather, you're modifying `childObj`, and childObj isn't a proxy. Becuase `proxify` doesn't proxyify non-enumerable props. That's why the added prop must be an object (or array).

Reply:
> It fixes it, but what's the point of adding a variable that is constant? I guess in most cases it is an oxymoron.

The constant object still has modifiable properties. You just can't replace the object itself.

```
Object.defineProperty(palindrom.obj, "myLocalNonSyncedObject", {
    enumerable: false,
    value: {},
    writable: false
});
palindrom.obj.myLocalNonSyncedObject.foo = "bar"; //this works
palindrom.obj.myLocalNonSyncedObject = "bar"; //this is ignored
```

But I agree this only makes sense for objects.

Reply:
Well, as per the link you've shared, `writable` defaults to `false`.

Reply:
^ true that!

Reply:
Maybe just change to this:

> Note:  You should not replace the value of the non-enumerable property after it has been defined. Otherwise Palindrom will still detect the change. If you want to modify the unobserved values, you should declare the non-enumerable property as an object and only modify its children.

Reply:
Much better!

Reply:
Done